pkg/eventlog/multiinterceptor.go

```diff
@@ -8,26 +8,23 @@ type repeater struct {
 	interceptors []Interceptor
 }
 
-func (r *repeater) InterceptWithReturn(events *events.EventList) (*events.EventList, error) {
-	panic("implement me")
-}
-
-func (r *repeater) Intercept(events *events.EventList) error {
+func (r *repeater) Intercept(events *events.EventList) (*events.EventList, error) {
 
 	// Avoid nil dereference if Intercept is called on a nil *Recorder and simply do nothing.
 	// This can happen if a pointer type to *Recorder is assigned to a variable with the interface type Interceptor.
 	// Mir would treat that variable as non-nil, thinking there is an interceptor, and call Intercept() on it.
 	// For more explanation, see https://mangatmodi.medium.com/go-check-nil-interface-the-right-way-d142776edef1
 	if r == nil {
-		return nil
+		return events, nil
 	}
 
 	for _, i := range r.interceptors {
-		if err := i.Intercept(events); err != nil {
-			return err
+		_, err := i.Intercept(events)
+		if err != nil {
+			return events, err
```

Comment:
By convention, we usually return the zero value (`nil` in this case) when an error occurs. I think here `nil` would be better too.